### PR TITLE
Convert submission insertion and grading selection sprocs to lib

### PR DESCRIPTION
**Pull request by jonatanschroeder**

### CODE REVIEW

File: apps/prairielearn/src/lib/question-variant.sql [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  variants AS v
+WHERE
+  v.instance_question_id = $instance_question_id;
+
+-- BLOCK insert_workspace
+INSERT INTO
+  workspaces DEFAULT
+VALUES
+RETURNING
+  id;
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/lib/question-variant.sql:31]: https://github.com/sql-formatter-org/sql-formatter/issues/718

Comment (nwalters512) [reply apps/prairielearn/src/lib/question-variant.sql:31]: I opened https://github.com/sql-formatter-org/sql-formatter/pull/720 to improve this.

File: apps/prairielearn/src/lib/question-variant.ts [hunk 2]
```diff
@@ -180,26 +195,100 @@ async function makeAndInsertVariant(
...
-      authn_user_id,
-      group_work,
-      require_open,
-      variant_course.id,
-      client_fingerprint_id,
-    ],
-    VariantWithFormattedDateSchema,
-  );
+
+  const variant = await sqldb.runInTransactionAsync(async () => {
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/question-variant.ts:199]: I can't comment on the line, but: the `group_work` parameter to `makeAndInsertVariant` is now unused (and was in fact unused before - we passed it to the sproc, but the sproc did nothing with it). Consider removing to clean that up.

**Review by nwalters512 — APPROVED** [review]
All these changes to core code make me nervous, but I went through this pretty carefully and this all looks good. Let's hope our tests catch all the scenarios we care about!

File: apps/prairielearn/src/lib/question-variant.ts (hunk 2 above)
Comment (jonatanschroeder) [reply apps/prairielearn/src/lib/question-variant.ts:197]: Done in 79ff2ab9e1e4c0b6b3f0bb2e15c3beb9fa466f47. This actually cascaded into multiple functions that also only had that parameter to pass it along. I'll do another passthrough to ensure everything is fine before merging.

Comment (jonatanschroeder) [reply apps/prairielearn/src/lib/question-variant.ts:197]: It was 5 functions, most of them local, so not too much.

**Pull request merged**